File: Data Structures & Algorithms/validate-parentheses/submission-5.java
class Solution {
    public boolean isValid(String s) {
        if(s.length()<=1) return false;
        Stack<Character> stack = new Stack<>();
        for(char c: s.toCharArray()){
            if(c=='[' || c=='{' || c=='(' ){
                stack.push(c);
            }else{
                if(stack.size()==0) return false;
                if((c==']' && stack.pop()!='[') || (c=='}' && stack.pop()!='{') || (c==')' && stack.pop()!='(')){
                    return false;
                }
            }
        }
        if(stack.size()>0) return false;
        return true;
    }
}

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-5.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack=[]
        ops={"+","-","/","*"}
        for i in tokens:
            if i in ops:
                # ops
                num2=stack.pop()
                num1=stack.pop()
                if i=="-":
                    stack.append(num1-num2)
                elif i=="*":
                    stack.append(num1*num2)
                elif i=="/":
                    stack.append(int(num1 / num2))
                else:
                    stack.append(num1+num2)
            else:
                stack.append(int(i))
        return stack[-1]